newIDE/app/src/ResourcesList/FileToCloudProjectResourceUploader.js

```diff
@@ -0,0 +1,206 @@
+          authenticatedUser,
+          cloudProjectId,
+          selectedFiles,
+          (current: number, total: number) => {
+            setUploadProgress((current / total) * 100);
+          }
+        );
+        const erroredResults = results.filter(({ error }) => !!error);
+        const okResults = results.filter(({ url }) => !!url);
+        if (erroredResults.length) {
+          throw erroredResults[0];
+        } else if (okResults.length) {
+          onChooseResources(
+            okResults.map(({ url, resourceFile }) => {
+              const newResource = createNewResource();
+              newResource.setFile(url || '');
+              newResource.setName(resourceFile.name);
+              newResource.setOrigin('cloud-project-resource', url || '');
+
+              return newResource;
+            })
+          );
+        }
+      } catch (error) {
+        showErrorBox({
+          message:
+            'There was an error while uploading some resources. Verify your internet connection or try again later.',
+          rawError: error,
+          errorId: 'upload-cloud-project-resource-error',
+          doNotReport: true,
+        });
+      } finally {
+        setIsUploading(false);
+      }
+    },
+    [
+      selectedFiles,
+      authenticatedUser,
+      onChooseResources,
+      createNewResource,
+      cloudProjectId,
+    ]
+  );
+
+  const invalidFiles = selectedFiles
+    .map(file => {
+      if (file.size > PROJECT_RESOURCE_MAX_SIZE_IN_BYTES) {
+        return {
+          filename: file.name,
+          error: 'too-large',
+        };
+      }
+      return null;
+    })
+    .filter(Boolean);
+
+  const canUploadWithThisStorageProvider =
+    storageProvider.internalName === 'Cloud';
+  const isConnected = !!authenticatedUser.profile;
+  const canChooseFiles =
+    !isUploading && isConnected && canUploadWithThisStorageProvider;
+  const canUploadFiles =
+    !isUploading &&
+    canChooseFiles &&
+    hasSelectedFiles &&
+    invalidFiles.length === 0;
+
+  return (
+    <ColumnStackLayout noMargin>
+      {!isConnected ? (
+        <AlertMessage kind="warning">
+          <Trans>
+            Your need to first create your account, or login, to upload your own
+            resources.
+          </Trans>
+        </AlertMessage>
+      ) : !canUploadWithThisStorageProvider ? (
+        <AlertMessage kind="warning">
+          <Trans>
+            Your need to first save your game on GDevelop Cloud to upload your
+            own resources.
+          </Trans>
+        </AlertMessage>
+      ) : null}
+      <Paper variant="outlined">
+        <Line expand>
+          <Column expand>
+            <input
+              accept={resourceKindToInputAcceptedFiles[options.resourceKind]}
+              style={styles.fileInput}
+              multiple={options.multiSelection}
+              type="file"
+              ref={inputRef}
+              disabled={!canChooseFiles}
```

Comment:
Same here, should it be disabled when uplaoding?

Reply:
It is ;)
`const canChooseFiles = !isUploading && isConnected && canUploadWithThisStorageProvider`